lib/model/query/entities.js

```diff
@@ -86,6 +87,49 @@ createNew.audit = (newEntity, dataset, partial, subDef) => (log) => {
 };
 createNew.audit.withResult = true;
 
+const _createMany = (dataset, entities, sourceId, userAgentIn) => async ({ all, context }) => {
+  const creatorId = context.auth.actor.map((actor) => actor.id).orNull();
+  const userAgent = blankStringToNull(userAgentIn);
+
+  const entityInsert = sql.join(entities.map(e => sql`(${sql.join([dataset.id, e.uuid, creatorId, sql`clock_timestamp()`], sql`,`)} )`), sql`,`);
+  const newEntities = await all(sql`
+  INSERT INTO entities ("datasetId", "uuid", "creatorId", "createdAt")
+  VALUES ${entityInsert}
+  RETURNING id`);
+
+  const defInsert = sql.join(entities.map((e, i) => sql`(${sql.join([
+    newEntities[i].id,
+    e.def.label,
+    JSON.stringify(e.def.data),
+    JSON.stringify(e.def.dataReceived),
+    sourceId,
+    creatorId,
+    userAgent,
+    'true',
+    'true',
+    sql`clock_timestamp()`,
+    '1'
+  ], sql`,`)} )`), sql`,`);
+
+  const defs = await all(sql`
+  INSERT INTO entity_defs ("entityId", "label", "data", "dataReceived",
+    "sourceId", "creatorId", "userAgent", "root", "current", "createdAt", "version")
+  VALUES ${defInsert}
+  RETURNING *
+  `);
+
+  return defs;
+};
+
+const createMany = (dataset, entities, sourceId, userAgentIn) => (container) =>
+  container.db.transaction((trxn) =>
+    container.with({ db: trxn }).Entities._createMany(dataset, entities, sourceId, userAgentIn))
+    .catch((err) => reject(err));
+
+createMany.audit = (dataset, entities, sourceId) => (log) =>
+  log('entity.bulk.create', dataset, { sourceId, count: entities.length });
```

Comment:
I think this should be about an entity (rather than a dataset) because while it does change the overall dataset by growing the dataset, it is more about entities. Also, it is the dataset actee ID, so there is already that link. 

Maybe it should be `entity.create.bulk` instead of `entity.bulk.create`?